lib/logstash/inputs/redis.rb

```diff
@@ -123,9 +123,9 @@ def run(output_queue)
     # ignore and quit
   end # def run
 
-  def teardown
-    @shutdown_requested = true
-    @teardown_method.call
+  def stop
+    @stop_requested = true
```

Comment:
when calling redefining stop you need to call super as this invokes code in the base class that is need.